From cead98e64b3529a72b93d85b9f8e3ad4381e448a Mon Sep 17 00:00:00 2001
From: Ian Lin <ian.lin-ee@infineon.com>
Date: Thu, 31 Mar 2022 01:10:38 -0500
Subject: [PATCH 119/137] brcmfmac: do not disable controller in apmode stop

Wpa_supplicant will firstly call scan before start apmode.
In mbss case, controller will be disabled during stop apmode
so next time start apmpde will fail.
Fix by removing the disable flow.

Signed-off-by: Ian Lin <ian.lin@infineon.com>



---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index fbfabe6468bd..d4abd7b4fd30 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -5464,11 +5464,6 @@ static int brcmf_cfg80211_stop_ap(struct wiphy *wiphy, struct net_device *ndev)
 			return 0;
 		}
 
-		if (ifp->vif->mbss) {
-			err = brcmf_fil_cmd_int_set(ifp, BRCMF_C_DOWN, 1);
-			return err;
-		}
-
 		/* First BSS doesn't get a full reset */
 		if (ifp->bsscfgidx == 0)
 			brcmf_fil_iovar_int_set(ifp, "closednet", 0);
-- 
2.17.1

